watsononlinestore/watson_online_store.py

```diff
@@ -292,6 +294,22 @@ def get_discovery_response(self, input_text):
             query_options={'query': input_text, 'count': DISCOVERY_QUERY_COUNT}
         )
 
+        # Watson discovery assigns a confidence level to each result.
+        # Based on data mix, we can assign a minimum tolerance value in an
+        # attempt to filter out the "weakest" results.
+        if self.discovery_score_filter:
+            filtered_results = []
+            num_matches = 0
+            if 'results' in discovery_response:
+                for result in discovery_response['results']:
```

Comment:
I don't think that helps.
First, I think the plural is off.  It should be:
results = (result for result in discovery_response['results']...
Second, you'd us that when the "if" appies to each result. No point in checking for 'results' inside the loop.
Also I do like trying to use list comprehension to make the code pythonic, but often after it is done, it looks like a new list was built when the original one was OK. So it looks less efficient and probably less readable.